src/nvim/testdir/test_signs.vim

```diff
@@ -1742,3 +1744,117 @@ func Test_sign_cursor_position()
   call StopVimInTerminal(buf)
   call delete('XtestSigncolumn')
 endfunc
+
+" Return the 'len' characters in screen starting from (row,col)
+func s:ScreenLine(row, col, len)
+  let s = ''
+  for i in range(a:len)
+    let s .= nr2char(screenchar(a:row, a:col + i))
+  endfor
+  return s
+endfunc
+
+" Test for 'signcolumn' set to 'number'.
+func Test_sign_numcol()
+  new
+  call append(0, "01234")
+  " With 'signcolumn' set to 'number', make sure sign is displayed in the
+  " number column and line number is not displayed.
+  set numberwidth=2
+  set number
+  set signcolumn=number
+  sign define sign1 text==>
+  sign place 10 line=1 name=sign1
+  sign define sign2 text=Ｖ
+  redraw!
+  call assert_equal("=> 01234", s:ScreenLine(1, 1, 8))
+
+  " With 'signcolumn' set to 'number', when there is no sign, make sure line
+  " number is displayed in the number column
+  sign unplace 10
+  redraw!
+  call assert_equal("1 01234", s:ScreenLine(1, 1, 7))
+
+  " Disable number column. Check whether sign is displayed in the sign column
+  set numberwidth=4
+  set nonumber
+  sign place 10 line=1 name=sign1
+  redraw!
+  call assert_equal("=>01234", s:ScreenLine(1, 1, 7))
+
+  " Enable number column. Check whether sign is displayed in the number column
+  set number
+  redraw!
+  call assert_equal("=>  01234", s:ScreenLine(1, 1, 9))
```

Comment:
For some reason there seems to be one too many space.

https://ci.appveyor.com/project/neovim/neovim/builds/34075588/job/ao12eot4yxwssrip#L13543
```
From test_signs.vim:
Found errors in Test_sign_numcol():
function RunTheTest[44]..Test_sign_numcol line 30: Expected '=>  01234' but got ' => 01234'
../../../build/bin/nvim.exe -u NONE -U NONE -i NONE --noplugin --headless -S summarize.vim messages
Test results:
```

Reply:
It seems that https://github.com/vim/vim/commit/4dff4aed09d2b0d570ca0d19de9cb08bdf03e695 is missing. I don't think this is necessary, but it seems to work differently with or without it compared to vim.

Reply:
OK.  I have changed the test code.